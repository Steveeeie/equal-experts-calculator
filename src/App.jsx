import React from 'react';
import { GlobalStyles } from './GlobalStyles';
import { Calculator } from './Calculator';

function App() {
  return (
    <>
      <GlobalStyles />
      <Calculator />
    </>
  );
}

export { App };
