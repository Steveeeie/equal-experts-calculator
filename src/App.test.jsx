import { render } from '@testing-library/react';
import { App } from './App';

describe('App', () => {
  test('renders the calculator', () => {
    const { getByTestId } = render(<App />);
    expect(getByTestId('calculator')).toBeTruthy();
  });

  test('applies global styles', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
