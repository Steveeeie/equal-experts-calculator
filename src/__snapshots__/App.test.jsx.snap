// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App applies global styles 1`] = `
.c1 {
  display: block;
  margin: 0 auto;
  width: 100px;
}

.c0 {
  background-color: #222222;
  border-radius: 16px;
  max-width: 320px;
  overflow: hidden;
  padding-top: 32px;
  width: 100%;
}

.c3 {
  font-size: 40px;
  line-height: 72px;
  overflow-x: auto;
  padding: 0 24px;
  text-align: right;
  white-space: nowrap;
}

.c2 {
  font-size: 20px;
  overflow-x: auto;
  padding: 24px 24px 0 24px;
  text-align: right;
  white-space: nowrap;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: 600;
  outline: none;
  padding: 16px 0;
  background-color: #333333;
}

.c5:hover {
  background-image: linear-gradient(rgba(255,255,255,0.1),rgba(255,255,255,0.1));
}

.c5:focus-visible {
  box-shadow: inset 0 0 0 2px white;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: 600;
  outline: none;
  padding: 16px 0;
  background-color: #fc6c12;
}

.c7:hover {
  background-image: linear-gradient(rgba(255,255,255,0.1),rgba(255,255,255,0.1));
}

.c7:focus-visible {
  box-shadow: inset 0 0 0 2px white;
}

.c8 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: 600;
  outline: none;
  padding: 16px 0;
  background-color: #444444;
}

.c8:hover {
  background-image: linear-gradient(rgba(255,255,255,0.1),rgba(255,255,255,0.1));
}

.c8:focus-visible {
  box-shadow: inset 0 0 0 2px white;
}

.c9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: 600;
  outline: none;
  padding: 16px 0;
  background-color: #1795d4;
  grid-column: span 2;
}

.c9:hover {
  background-image: linear-gradient(rgba(255,255,255,0.1),rgba(255,255,255,0.1));
}

.c9:focus-visible {
  box-shadow: inset 0 0 0 2px white;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 25%;
}

.c4 {
  display: grid;
  grid-template-columns: repeat(4,1fr);
  gap: 1px;
}

<div>
  <div
    class="c0"
    data-testid="calculator"
  >
    <img
      alt="Equal Experts"
      class="c1"
      src="logo.svg"
    />
    <div
      class="c2"
      data-testid="previous-equation"
    />
    <div
      class="c3"
      data-testid="current-equation"
    >
      0
    </div>
    <div
      class="c4"
    >
      <button
        class="c5"
        data-testid="button-ce"
      >
        CE
      </button>
      <button
        class="c5"
        data-testid="button-ac"
      >
        AC
      </button>
      <span
        class="c6"
      />
      <button
        class="c7"
        data-testid="button-divide"
      >
        /
      </button>
      <button
        class="c8"
        data-testid="button-7"
      >
        7
      </button>
      <button
        class="c8"
        data-testid="button-8"
      >
        8
      </button>
      <button
        class="c8"
        data-testid="button-9"
      >
        9
      </button>
      <button
        class="c7"
        data-testid="button-times"
      >
        X
      </button>
      <button
        class="c8"
        data-testid="button-4"
      >
        4
      </button>
      <button
        class="c8"
        data-testid="button-5"
      >
        5
      </button>
      <button
        class="c8"
        data-testid="button-6"
      >
        6
      </button>
      <button
        class="c7"
        data-testid="button-minus"
      >
        -
      </button>
      <button
        class="c8"
        data-testid="button-1"
      >
        1
      </button>
      <button
        class="c8"
        data-testid="button-2"
      >
        2
      </button>
      <button
        class="c8"
        data-testid="button-3"
      >
        3
      </button>
      <button
        class="c7"
        data-testid="button-plus"
      >
        +
      </button>
      <button
        class="c8"
        data-testid="button-0"
      >
        0
      </button>
      <button
        class="c8"
        data-testid="button-decimal"
      >
        .
      </button>
      <button
        class="c9"
        data-testid="button-equals"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
